app/javascript/vue/repository_item_sidebar/Textarea.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <textarea v-if="editing"
+            ref="textareaRef"
+            class="leading-5 inline-block outline-none border-solid font-normal border-[1px] box-content
+                  overflow-x-hidden overflow-y-auto resize-none rounded px-4 py-2 w-[calc(100%-2rem)]
+                  border-sn-science-blue"
+            :class="{
+              'max-h-[4rem]': collapsed,
+              'max-h-[40rem]': !collapsed
+            }"
+            :placeholder="placeholder"
+            v-model="value"
+            @keydown="handleKeydown"
+            @blur="handleBlur" />
+  <div v-else
+        :ref="unEditableRef"
+        class="grid box-content sci-cursor-edit font-normal border-solid px-4 py-2 border-sn-light-grey rounded
+              leading-5 border outline-none hover:border-sn-sleepy-grey overflow-y-auto whitespace-pre-line
+              w-[calc(100%-2rem)]"
+        :class="{ 'max-h-[4rem]': collapsed,
+                  'max-h-[40rem]': !collapsed, }"
+        @click="enableEdit">
+    <span v-if="smartAnnotation"
+          v-html="sa_value || noContentPlaceholder"
+          class="[&>p]:mb-0"></span>
+    <span v-else>{{ value || noContentPlaceholder }}</span>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Textarea",
+  data() {
+    return {
+      value: '',
+      editing: false,
+    };
+  },
+  props: {
+    expandable: { type: Boolean, required: true },
+    collapsed: { type: Boolean, required: true },
+    initialValue: String,
+    noContentPlaceholder: String,
+    placeholder: String,
+    decimals: { type: Number, default: null },
```

Comment:
I would add another more explicit property, like isNumber, to base the number enforcement on